R/jointRPCA.R

```diff
@@ -0,0 +1,752 @@
+#'   \item \code{.joint_optspace_helper()} — internal coordination of multi-view OptSpace factorization.
+#'   \item \code{.transform()} and \code{.transform_helper()} — projection of new data into existing ordination space.
+#'   \item \code{.rpca_table_processing()} — compositional preprocessing (filtering, zero-removal, etc.).
+#'   \item \code{.mask_value_only()} — safe numeric matrix masking for missing values.
+#'   \item Other small utility functions used across the Joint-RPCA workflow.
+#' }
+#'
+#' These functions depend on the OptSpace backend defined in \code{optspace.R}.
+#' Only \code{jointRPCAuniversal()} and \code{runJointRPCA()} are exported; all others are internal helpers.
+#'
+#' @keywords internal
+#' @noRd
+NULL
+
+#' Universal Joint RPCA Wrapper
+#'
+#' @param x Input object: \code{MultiAssayExperiment}, \code{SummarizedExperiment}
+#'   (including \code{TreeSummarizedExperiment}), list of matrices, or single matrix.
+#' @param experiments Character vector of experiment names to extract when \code{x}
+#'   is a \code{MultiAssayExperiment} (i.e. \code{names(experiments(x))}).
+#'   If \code{NULL}, all experiments are used.
+#'
+#'   For \code{MultiAssayExperiment} inputs, \strong{one assay per experiment} is
+#'   used: by default the first assay returned by
+#'   \code{SummarizedExperiment::assayNames()} (or index \code{1L} if unnamed).
+#'   The actually used assay names are recorded in \code{$assay.names.used} in
+#'   the result. If you need a different assay (e.g. \code{"relab"} instead of
+#'   \code{"counts"}), subset or reorder assays in \code{x} before calling
+#'   \code{jointRPCAuniversal()}.
+#' @param ... Additional arguments passed to \code{.jointRPCA()}.
+#'
+#' @return Output from \code{.jointRPCA()} with extra fields when \code{x} is a
+#'   \code{MultiAssayExperiment}:
+#'   \itemize{
+#'     \item \code{$experiment.names}: character vector of experiments used.
+#'     \item \code{$assay.names.used}: named character vector giving, for each
+#'           experiment, the assay name that was used (typically the first in
+#'           \code{assayNames()}).
+#'   }
+#' @export
+
+jointRPCAuniversal <- function(x, experiments = NULL, ...) {
+    assay_names_used <- NULL
+    
+    if (inherits(x, "MultiAssayExperiment")) {
+        
+        exps <- MultiAssayExperiment::experiments(x)
+        
+        if (is.null(experiments)) {
+            experiments <- names(exps)
+        }
+        if (length(experiments) == 0L) {
+            stop("No experiments found in 'x'.")
+        }
+        
+        assay_names_used <- setNames(
+            character(length(experiments)),
+            experiments
+        )
+        
+        tables <- lapply(experiments, function(e) {
+            exp_se <- exps[[e]]
+            if (is.null(exp_se)) {
+                stop(sprintf("Experiment '%s' not found in 'x'.", e))
+            }
+            
+            anm <- SummarizedExperiment::assayNames(exp_se)
```

Comment:
If possible, use function names without specifying the package:
`
SummarizedExperiment::assayNames()` --> `assayNames()`